starter_code/server.js

```diff
@@ -1,30 +1,66 @@
 import express from "express"
-import mongoose from "mongoose"
 import "./db-connect.js" // connect to database
-import { Customer } from "./models.js"
+import Customer from "./models/Customer.js"
+import Order from "./models/Order.js"
+import Pizza from "./models/Pizza.js"
 
 const app = express()
 
-
-// STARTP API
-const port = 5000
-app.listen(port, () => {
-    console.log('Server started on port ' + port);
-});
-
 // ROUTES
 app.get('/', (req, res) => {
-    res.send("<h1>Hello from Pizza API</h1>");
+    res.send(`<h1>Hello from Pizza API</h1>
+    <p><a href="/customers">/customers</a></p>
+    <p><a href="/orders">/orders</a></p>
+    <p><a href="/pizzas">/pizzas</a></p>`);
 });
 
-app.get('/customers', async (req, res) => {
-  const customers = [] // please fetch the customers from your database here, por favor!
-  res.json(customers) 
+app.get('/customers', async (req, res, next) => {
+    try {
+        const customers = await Customer.find()
+        if (!customers) throw new Error(400, "No customers found");
+        res.json(customers)
+    } catch (error) {
+        next(error)
+    }
+})
+
+app.get('/orders', async (req, res, next) => {
+    try {
+        const ordersWithUsersAndPizzas = await Order.find()
+        .select("-_id")
+        .populate({path: 'customer', model: Customer, select: '-_id address.city'})
+        .populate({path: 'items.pizza', model: Pizza, select: '-_id'})
```

Comment:
Yes! That is the trick.
By the way: Populate does not need to get told the model, if you have placed the "ref" attribute.
So `.populate("items.pizza", "-_id")` would do the same :)